From b54aef86c6c9ef654da8ba6c06bfc049ced6af0b Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Wed, 28 Jan 2026 11:35:20 +0100
Subject: x86/hyperv: Fix smp_ops build failure on UP kernels
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

From: Ingo Molnar <mingo@kernel.org>

[ Upstream commit ac059ae422d7d05ed9d62970a30fa3b95870b967 ]

CI testing found this build failure:

  arch/x86/hyperv/hv_crash.c:631:9: error: ‘smp_ops’ undeclared (first use in this function)

And I bisected it back to the initial commit that enabled this feature:

  77c860d2dbb72d1f3c6a2e882a07d19eca399db5 is the first bad commit
  commit 77c860d2dbb72d1f3c6a2e882a07d19eca399db5 (HEAD)
  Author: Mukesh Rathor <mrathor@linux.microsoft.com>
  Date:   Mon Oct 6 15:42:08 2025 -0700

  x86/hyperv: Enable build of hypervisor crashdump collection files

Hyperv should probably be limited to SMP kernels, as nobody
appears to be testing it on UP kernels.

Until then, fix the smp_ops assumption. Build tested only.

Fixes: 77c860d2dbb72 ("x86/hyperv: Enable build of hypervisor crashdump collection files")
Cc: Mukesh Rathor <mrathor@linux.microsoft.com>
Cc: Wei Liu <wei.liu@kernel.org>
Cc: linux-kernel@vger.kernel.org
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 arch/x86/hyperv/hv_crash.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/hyperv/hv_crash.c b/arch/x86/hyperv/hv_crash.c
index c0e22921ace1a..a78e4fed57203 100644
--- a/arch/x86/hyperv/hv_crash.c
+++ b/arch/x86/hyperv/hv_crash.c
@@ -628,7 +628,9 @@ void hv_root_crash_init(void)
 	if (rc)
 		goto err_out;
 
+#ifdef CONFIG_SMP
 	smp_ops.crash_stop_other_cpus = hv_crash_stop_other_cpus;
+#endif
 
 	crash_kexec_post_notifiers = true;
 	hv_crash_enabled = true;
-- 
2.51.0

